pkg/telemetryquerylanguage/contexts/tqlmetrics/metrics.go

```diff
@@ -25,31 +25,31 @@ import (
 	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/telemetryquerylanguage/tql"
 )
 
-type MetricTransformContext struct {
+type TransformContext struct {
```

Comment:
Why is the "item" a data point and not a metric?

Reply:
Most of the functions apply per DataPoint instead of per metric. T[hats discussed briefly in the processing doc.](https://github.com/open-telemetry/opentelemetry-collector/blob/main/docs/processing.md#telemetry-query-language) The datapoint's metric is exposed as a "virtual field" similar to Resource and InstrumentationScope.